### PR TITLE
Limit Max Window Size

### PR DESCRIPTION
**Pull request by pschoffer**
### Plugin Updated: Search

`index.max_result_window` is a setting that controls the maximum number of documents that can be paginated through. This change limits the maximum number to be 9000.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples. 

## Steps to Test

1. Run `curl http://localhost:49158/vip-200508-post-1-v2/_settings -s | jq | grep max_result` 
1. After applying the change and putting new mappings `wp vip-search put-mappings`
1. Run `curl http://localhost:49158/vip-200508-post-1-v2/_settings -s | jq | grep max_result`

### CODE REVIEW

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]
Thank you, looks good! 

I left one minor suggestion - instead of multiple returns, we can use a `return min()` which would return a minimum value out of the list of passed arguments.

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -484,6 +485,12 @@ protected function setup_hooks() {
 
 		add_filter( 'vip_search_post_meta_allow_list', array( $this, 'filter__vip_search_post_meta_allow_list_defaults' ) );
 
+		// Limit the max result window
+		add_filter( 'ep_max_result_window', [ $this, 'limit_max_result_window' ], PHP_INT_MAX );
```
Comment (nickdaugherty) [inline search/includes/classes/class-search.php:489]: A quick inline comment that indicates which one of these is for the index settings and which one is for the query parameters would be helpful for our future selves.

File: search/includes/classes/class-search.php [hunk 2]
```diff
@@ -1738,4 +1745,13 @@ public function is_protected_content_enabled() {
...
 		return $protected_content_feature->is_active();
 	}
+
+	public function limit_max_result_window( $current_value ) {
+
+		if ( $current_value > self::MAX_RESULT_WINDOW ) {
+			return self::MAX_RESULT_WINDOW;
+		}
+
+		return $current_value;
```
[the remark this answers is not in the document]
Comment (pschoffer) [reply search/includes/classes/class-search.php]: nice!

File: search/includes/classes/class-search.php (hunk 1 above)
Comment (pschoffer) [reply search/includes/classes/class-search.php:489]: Nice idea!

**Review by nickdaugherty — APPROVED** [review]

**Comment by rinatkhaziev** [comment]
r1480-stacks

**Pull request merged**